src/compiler/checker.ts

```diff
@@ -29886,7 +29889,7 @@ namespace ts {
 
         function checkExpressionForMutableLocation(node: Expression, checkMode: CheckMode | undefined, contextualType?: Type, forceTuple?: boolean): Type {
             const type = checkExpression(node, checkMode, forceTuple);
-            return isConstContext(node) ? getRegularTypeOfLiteralType(type) :
+            return isConstContext(node) || isCommonJsExportedExpresion(node) ? getRegularTypeOfLiteralType(type) :

```

Comment:
Should `isConstContext` just include `isCommonJsExportedExpression`?

Reply:
Naw, `isConstContext` does deep-const'ing, which we do not want.